TFNetworkLayer.py

```diff
@@ -5516,9 +5520,12 @@ def init(self, output, output_with_activation=None, target=None, layer=None):
       if self.output.have_time_axis():
         self.output_seq_lens = output.get_sequence_lengths()
         time_and_batch_dims = (self.output.time_dim_axis, self.output.batch_dim_axis)
-        assert time_and_batch_dims in [(0, 1), (1, 0)], "output time-batch-dim unexpected: %s" % self.output
+        assert time_and_batch_dims in [(0, 1), (1, 0), (2, 0)], "output time-batch-dim unexpected: %s" % self.output
+        if self.output.time_dim_axis == 2:
+          self.output = self.output.copy_with_feature_dim_axis(-1)
```

Comment:
If you anyway do this, then why don't you do it right in the beginning, **before** `time_and_batch_dims = (...)`? Then you don't need to change any other code, and it's much simpler. So please do that.

Reply:
Also, instead of checking `time_dim_axis == 2`, check for `is_batch_feature_major`.